controllers/deactivation/predicate.go

```diff
@@ -25,3 +29,58 @@ func (CreateAndUpdateOnlyPredicate) Delete(_ event.DeleteEvent) bool {
 func (CreateAndUpdateOnlyPredicate) Generic(_ event.GenericEvent) bool {
 	return false
 }
+
+type GenerationOrConditionsChangedPredicate struct {
+	predicate.GenerationChangedPredicate
+}
+
+// Update implements default UpdateEvent filter for validating no generation change
```

Comment:
Not only the generation change. It also takes into account the conditions.
Could you please fix the description and also add a description to `type GenerationOrConditionsChangedPredicate struct {` with an overall definition of what this predicate does?

Reply:
Done